src/scroll-timeline-css-parser.js

```diff
@@ -354,6 +375,50 @@ export class StyleParser {
     }
   }
 
+  extractScrollTimelineNames(contents) {
+    const value = RegexMatcher.ANIMATION_TIMELINE.exec(contents)[1].trim();
+    const timelineNames = [];
+
+    value.split(",").map(part => part.trim()).forEach(part => {
+      if(isAnonymousScrollTimeline(part)) {
+        const name = this.saveAnonymousTimelineName(part);
+        timelineNames.push(name);
+      } else {
+        timelineNames.push(part);
+      }
+    });
+
+    return timelineNames;
+  }
+
+  saveAnonymousTimelineName(part) {
+    // Anonymous scroll timelines are given a name that starts with ':' to
+    // prevent collision with named scroll timelines.
+    const name = `:t${this.anonymousScrollTimelineOptions.size}`;
```

Comment:
let nextAnonymousTimelineNameIndex = 0;

saveAnonymousTimelineName(part) {
   const name = `:t${nextAnonymousTimelineNameIndex++}`;
   ...
}

Presently, we have a "memory leak" in that we don't release the name / timeline association when the timeline is no longer needed.  Once we fix the leak, we then introduce the possibility of reusing the name since ...Options.size could shrink and then grow again.